vrs/marten-construction-script.py

```diff
@@ -0,0 +1,101 @@
+final_10k_indel = final_10k_indel.sample(indel_sample_freq)
+
+final_10k_longRef = ht_whole.filter(hl.len(ht_whole.alleles[0])>3)
+longRef_sample_freq = 15000/(759000000*ref_freq)
+final_10k_longRef = final_10k_longRef.sample(longRef_sample_freq)
+
+final_10k_longVar = ht_whole.filter(hl.len(ht_whole.alleles[1])>3)
+longVar_sample_freq = 15000/(759000000*var_freq)
+final_10k_longVar = final_10k_longVar.sample(longVar_sample_freq)
+
+final_9k_x = ht_whole.filter(ht_whole.locus.contig == "chrX")
+x_sample_freq = 9500/(759000000*x_freq)
+final_9k_x = final_9k_x.sample(x_sample_freq)
+
+final_1k_y = ht_whole.filter(ht_whole.locus.contig == "chrY")
+y_sample_freq = 1500/(759000000*y_freq)
+final_1k_y = final_1k_y.sample(y_sample_freq)
+
+# combining these tables into one final Hail Table to work with 
+
+tableParts = [final_50k_rand, final_10k_indel, final_10k_longRef, final_10k_longVar, final_9k_x, final_1k_y]
+final_struct = tableParts[0]
+for item_to_union in tableParts[1:]:
+    final_struct = final_struct.union(item_to_union) # final table of all but the multiallelic sites
+
+# TEMPORARY PATH BELOW PLEASE USE YOUR OWN PATH 
+# final_struct = final_struct.checkpoint("gs://seqr-scratch-temp/marten-temp-01-04/non-multiallelic-checkpoint.ht")
+fs = final_struct.count()
+print(f'Samples before monoallelic spike-in: {fs}')
+# 01-04 @ 5:08 pm: only ~79,000 spiked-in when I wanted 90,000 , probably lots of overlap 
+# between random and the extra ones 
+
+# constructing a table of only multiallelic sites
+
+ht1 = ht_whole._filter_partitions(sample(range(ht_whole.n_partitions()), 4)) # reads in 4 partitions (~15k var)
+ht1_gb = ht1.group_by(ht1.locus).aggregate(n=hl.agg.count()) # gb for group-by
+ht1_filt = ht1_gb.filter(ht1_gb.n > 3) # only the sites for multiallelic sites
+# -- 
+multiAlleleCount = ht1_filt.count()
+print(f'There are {multiAlleleCount} very multiallelic SITES in the table above') # 2292 sites, plenty of variables
+# -- 
+multiList = ht1_filt.locus.collect(_localize = False)
+ht_multiallelic = ht1.filter(multiList.contains(ht1.locus))
+# ^ would be awfully slow and expensive IF we weren't working with only 4 partiions
+
+# checkpoint for easier working PLEASE USE YOUR OWN PATH HERE 
+# ht_multiallelic = ht_multiallelic.checkpoint("gs://seqr-scratch-temp/marten-temp-01-04/multiallelic-checkpoint-v2.ht")# , overwrite=overwrite, _read_if_exists=read_if_exists)
+# ^^^ TEMPORARY PATH
+
+# multiallalic looking/summary
+
+ht_multiallelic.head(n=10).show()
+ht_multiallelic.tail(n=10).show()
+ht_multiallelic.count()
+
+# outputting! also doing this allows us to count the total size without it melting my computer 
+
+marten_final_output = final_struct.union(ht_multiallelic)
+
+# TEMPORARY PATH PLEASE USE YOUR OWN PATHS 
+# marten_final_output = marten_final_output.checkpoint("gs://seqr-scratch-temp/marten-temp-01-04/final-100k-01-04-23.ht")
+
```

Comment:
also just noting to add the code for exporting to a vcf since that's what will be handed off, and also include code for fixing the VCF header (add lines for the filters) so that the script doesn't error out on their end (can use this function: https://github.com/broadinstitute/gnomad_qc/blob/768d249cf02e1bc74614086d1fe098d69a622f01/gnomad_qc/v4/resources/release.py#L109)

Reply:
yes, this code would be good to add!